feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,566 @@
+            Proxy for the basis distribution.
+
+        d_meas: np.array.
+            Proxy for the measurement distribution.
+
+        Returns
+        -------
+        psi_value: float.
+            PSI value.
+        """
+        # Calculate the ratio of samples in each bin
+        basis_ratio = d_basis / d_basis.sum()
+        meas_ratio = d_meas / d_meas.sum()
+
+        # Necessary to avoid divide by zero and ln(0). Has minor impact on PSI value.
+        basis_ratio = np.where(
+            basis_ratio <= 0, self.min_pct_empty_buckets, basis_ratio
+        )
+        meas_ratio = np.where(meas_ratio <= 0, self.min_pct_empty_buckets, meas_ratio)
+
+        # Calculate the PSI value
+        psi_value = np.sum(
+            (meas_ratio - basis_ratio) * np.log(meas_ratio / basis_ratio)
+        )
+
+        return psi_value
+
+    def _split_dataframe(self, X):
+        """
+        Split a dataframe according to a cut-off value and return two dataframes:
+        one with values above the cut-off and one with values below or equal
+        to the cut-off.
+        The cut-off value is associated to a specific column.
+
+        The cut-off can be defined in two ways:
+
+            - Considering all observations. In that case a split fraction of 0.25
+            will ensure 25% of the observations are in the "below cut-off"
+            dataframe. The final number of observations in the "below cut-off" may
+            be slightly different than 25% as all observations with the same value
+            are put in the same split.
+
+            - Considering the values of the observations. The cut-off applies to
+            the distinct values. In that case a split fraction of 0.25 will ensure
+            25% of the distinct values are in the "below cut-off" dataframe;
+            regardless of the number of observations.
+
+        Parameters
+        ----------
+        X : pandas dataframe
+            The original dataset. Correspond either to the measurement dataset
+            or will be spit into a measurement and a basis dataframe.
+
+        Returns
+        -------
+        pandas dataframe with value below the
+        basis pandas dataframe
+        """
+        # Identify the values according to which the split must be done.
+        if self.split_col is None:
+            reference = pd.Series(X.index.to_list())
+        else:
+            reference = X[self.split_col]
+
+        # Raise an error if there are missing values in the reference column.
+        if reference.isna().sum() != 0:
+            raise ValueError(
+                "Na's are not allowed in the columns used to split the dataframe."
+            )
+
+        # Define the cut-off point based on quantile.
+        if self.split_distinct_value:
+            cut_off = self._get_cut_off_value(reference.unique())
+        else:
+            cut_off = self._get_cut_off_value(reference)
+
+        self.quantile = {self.split_col: cut_off}
```

Comment:
Questions:

I guess self.quantile is because we want to inform the user where the cut-off is made? Do we really need to expose this value?

If it is learned during fit, the name has to have a trailing underscore to abide by sklearn convention. 

Also, is quantile the best name? 

We only allow 1 cut-of column, so I don't think we need to return a dictionary, I would just return the value. 

We are not giving the user the option to divide the data based on a certain date. We should probably add that, and we should allow the user to pass a string with a date within the init

Reply:
I got up thinking about this today: at the moment, we have functionality to split data based on proportions. But thinking of the examples you gave me, we should probably also add:

data ==> cut when df[split_col] > data
numerical value ==> cut when df[split_col] > number (thinking of the contract ID, I may want to split after a certain ID
categorical value ==> cut when df[split_col].isin([category list]), thinking on a column that says portfolio1, portfolio2 and portfolio3, I may want portfolios 2 and 3 in the meas df.

Do these make sense?

Reply:
I am having second thoughts on whether to call the param split_frac or test_size. What 's your view?